apps/src/templates/Dialog.jsx

```diff
@@ -82,23 +85,34 @@ Cancel.propTypes = {
   children: React.PropTypes.node,
 };
 
-export function Buttons({children}) {
-  children = React.Children.toArray(children);
-  var leftChildren = children.slice(0, children.length-1);
-  var rightChild = children[children.length - 1];
-  return (
-    <div style={styles.buttons}>
-      <div style={styles.rightButton}>
-        {rightChild}
+export const Buttons = React.createClass({
+  getConfirmButton() {
+    return this.refs.buttons.getElementsByClassName('confirmButton')[0];
+  },
+
+  getCancelButton() {
+    return this.refs.buttons.getElementsByClassName('cancelButton')[0];
+  },
```

Comment:
Selecting via css class seems safe here since I'm only selecting from among children of `buttons`.